packages/babel-plugin-transform-async-to-promises/src/index.js

```diff
@@ -0,0 +1,112 @@
+/* Babel 7 transform */
+
+const babylon = require("babylon");
+const generate = require("@babel/generator").default;
+
+function printNode(ast) {
+  return generate(ast, {}).code;
+}
+
+function parse(code) {
+  return babylon.parse(code, {
+    allowImportExportEverywhere: true,
+    allowReturnOutsideFunction: true,
+    allowSuperOutsideMethod: true,
+  }).program;
+}
+
+const transform = require("nodent-transform").transform;
+
+function transformAsyncToPromises(api, options) {
+  let requiresTranspilation;
+  const runtime = options.runtime ? options.runtime : null;
+  let opts = {
+          // Code generation options
+          es6target: false,
+          babelTree: true,
+          engine: false,
+          generators: false,
+          promises: true,
+          lazyThenables: false,
+          wrapAwait: true,
+          noRuntime: !runtime,
+          $runtime: runtime,
+          generatedSymbolPrefix: "$",
+          $return: "$return",
+          $error: "$error",
+          $arguments: "$args",
+          $Promise: "Promise",
+          $asyncspawn: "$asyncspawn",
+          $asyncbind: "$asyncbind",
+          $makeThenable: "$makeThenable",
+        }; 
+
+  if (options.codeGenerationOptions) {
+      let keys = Object.keys(options.codeGenerationOptions);
+      for (let i=0; i<keys.length;i++) {
+	  if (keys[i] in opts)
+	      opts[keys[i]] = options.codeGenerationOptions[keys[i]];
+      }
+  }
+  
+  return {
+    visitor: {
+      Program: {
+        enter: function() {
+          requiresTranspilation = false;
+        },
+        exit: function(path, state) {
+          // Check if there was an async or await keyword before bothering to process the AST
+          if (!requiresTranspilation) return;
+
+          const newAst = transform(
+            {
+              // Input: the ast and filename
+              filename: state.filename,
+              ast: path.node,
+            },
+            opts,
+            {
+              // Helpers for the transformer:
+              parse: parse, // Parse a JS fragment into an AST
+              printNode: printNode, // Print a node as JS source
+              logger: false /* console.log.bind(console)*/, // Log a warning
+            },
+          ).ast;
+
+          if (runtime) {
+            newAst.body.splice(0, 0, {
+              type: "ImportDeclaration",
```

Comment:
Could you please use babel-types for that? and maybe babel-traverse to place to node at the correct location?

Reply:
Actually, I don't have the bandwidth to learn a new API or style/code conventions. I started this as a favour to @hzoo, and as far as I know the current implementation works. 

@hzoo - I'm happy to invite you (or anyone else) to my babel fork, but I'm really unable to provide the time to do this to the Babel team's satisfaction.